### PR TITLE
Update externals to what is expected to be in cesm2_3_beta17 fix some conda issues...

### LINKED ISSUE #1658: fsurdat_modifier creates a malformatted file on izumi
### Brief summary of bug

When trying to run fsurdat_modifier on izumi with the standard python module loaded, the resulting file cannot be read. ncdump -h gives: `ncdump: fsurdat.nc: fsurdat.nc: NetCDF: Unknown file format`.

### General bug information

**CTSM version you are using:** Branch a little past master (but I haven't changed the python code in this branch)

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** Just fsurdat_modifier on izumi

### Details of bug

I was hoping we could run the FSURDATMODIFIER test on izumi. So I first tried running `FSURDATMODIFYCTSM_D_Mmpi-serial_Ld1.5x5_amazon.I2000Clm50SpRs.izumi_intel`. This failed.

Then I loaded a module environment on izumi that is close to what is loaded by default for running CTSM / CESM on izumi:

  1) lang/python/3.7.0                2) mpi/2.3.3/intel/20.0.1           3) tool/hdf5/1.12.0/intel/20.0.1    4) tool/netcdf/4.7.4/intel/20.0.1   5) compiler/intel/20.0.1

I then tried running the tool manually: from the case directory of this test (so the .cfg file existed there), I first removed the previously-generated fsurdat.nc and then ran:

`python ~/ctsm_code/ctsm/tools/modify_fsurdat/fsurdat_modifier -v modify_fsurdat.cfg`.

This had the same result – even though I got a message saying that the surface dataset was successfully generated.

This is possibly (likely?) a system configuration issue – e.g., maybe the versions of xarray and other packages need to be updated on izumi??? It isn't critical that this is resolved; I'm okay with it being closed as a wontfix if others feel we don't care about supporting this tool on izumi.

@slevisconsulting - thoughts? Or @negin513 would you want to try this with your python environment on izumi?

### LINKED ISSUE #1910: modify_singlept_site_neon.py can't work on izumi, because of bad conda environment
### Brief summary of bug

modify_singlept_site_neon.py, fails to run on izumi, because the version of pandas in the conda environment when using py_env_create for what's used in python than what we've specified in the conda environment.

### General bug information

**CTSM version you are using:** ctsm5.1.dev114

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** running modify_singlept_site_neon on izumi

### Details of bug

conda claims that the version of pandas used is:

pandas                    1.3.5

But, internally pd.__version__ shows...

0.23.4

and pd.show_versions() also shows that pandas is older than what's used in the conda environment.

### Important details of your setup / configuration so we can reproduce the bug

I think this is really a problem of the conda environment in izumi more so than a problem with the python script itself. The method used requires at least version 1.1.0.

### Important output or errors that show the problem

```
Traceback (most recent call last):
  File "/fs/cgd/data0/erik/ctsm_worktree/neon/tools/site_and_regional/modify_singlept_site_neon.py", line 722, in <module>
    main()
  File "/fs/cgd/data0/erik/ctsm_worktree/neon/tools/site_and_regional/modify_singlept_site_neon.py", line 663, in main
    fill_interpolate(f2, "PCT_CLAY", method)
  File "/fs/cgd/data0/erik/ctsm_worktree/neon/tools/site_and_regional/modify_singlept_site_neon.py", line 504, in fill_interpolate
    tmp = tmp_df.to_numpy()
  File "/usr/local/anaconda-3.7.0/lib/python3.7/site-packages/pandas/core/generic.py", line 4376, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'to_numpy'
```

### LINKED ISSUE #2331: py_env_create fails on Derecho
### Brief summary of bug

As first noted by @olyson, calling `./py_env_create` on Derecho gives an error related to incompatible packages. This happens even if the conda environment name is changed from `ctsm_pylib` to `ctsm_pylib2` (or something else not matching an existing conda environment).

This can be resolved by removing version number specifiers from `numpy` and `numba`, but that may not be how we want to handle it.

### General bug information

**CTSM version you are using:** ctsm5.1.dev163

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** n/a

### Important output or errors that show the problem

```
LibMambaUnsatisfiableError: Encountered problems while solving:
  - package numba-0.55.2-py37h43839f2_0 requires numpy >=1.19.5,<2.0a0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ numba 0.55.2  is installable with the potential options
│  ├─ numba 0.55.2 would require
│  │  └─ numpy >=1.21.6,<2.0a0 , which can be installed;
│  └─ numba 0.55.2 would require
│     └─ numpy >=1.19.5,<2.0a0 , which can be installed;
└─ numpy 1.18.5  is not installable because it conflicts with any installable versions previously reported.

Trouble installing the ctsm_pylib python environment
There must be a problem in the ./python/conda_env_ctsm_py.txt conda specification environment file
Change the file and try again
```

### LINKED ISSUE #2351: Update to cesm2_3_beta17 externals
cesm2_3_beta17 is getting closer, and when it looks like it's at least known we can update to it's externals.

Right now that looks like:

cime6.0.217_httpsbranch01
ccs_config_cesm0.0.92
cmeps0.14.50
cdeps1.0.28
share1.0.18
cismwrap_2_1_97
rtm1_0_79
mosart1_0_49

At the same time we might as well update manage_externals to the latest version. It isn't strictly required for CTSM, but we could do it to help anyone who has branches or clones that use CAM where it is required.

### LINKED ISSUE #2380: ctsm_pylib on Izumi can now use Python 3.7.9
Perhaps not worth worrying about, but after the recent `conda` upgrade on Izumi I was wondering if the Python 3.7.0 specification is unnecessary. It turns out, if you specify Python 3.7.9 in `python/conda_env_ctsm_py_cgd.txt`, it now installs just fine. Everything in `make all` works except the stuff dependent on cdo, nco, and the inputdata directory.

### PR DESCRIPTION
**Pull request by ekluzek**
### Description of changes

Update externals to what's currently planned for cesm2_3_beta17.

Also update manage_externals. This is only important if you add CAM to your externals.
This was important on the Dust PR, so this will allow that PR to show less differences
to master when this comes in.

### Specific notes

Contributors other than yourself, if any:

CTSM Issues Fixed (include github issue #):
  Fixes #2351
  Fixes #2380
  Fixes #2331
  Fixes #1910
  Fixes #1658

Are answers expected to change (and if so in what way)? No

Any User Interface Changes (namelist or namelist defaults changes)? No

Testing performed, if any: standard
  ERI tests currently fail
  This one might have trouble: ERP_P256x2_Ld30.f45_f45_mg37.I2000Clm51FatesRs.derecho_intel.clm-mimicsFatesCold
  This one shows different answers so I'll fix that: SMS_D_Lm1_Mmpi-serial.CLM_USRDAT.I1PtClm50SpRs.derecho_intel.clm-USUMB_nuop

### CODE REVIEW

**Comment by ekluzek** [comment]
@samsrabin I added you as reviewer for this. The main thing to look at is py_env_create.

**Comment by ekluzek** [comment]
@glemieux can you try using this branch to create a conda environment with py_env_create to test if #2331 works for you for the LUH tools?

**Comment by ekluzek** [comment]
The change in answers is just because the USUMB test failed previously , and the FatesMimics test is expected to fail. So it's just the ERI tests that I need to figure out.

**Comment by ekluzek** [comment]
I now have a cime tag that fixes the ERI tests, so they are fine. So aux_clm should be good, but I need to rerun as I just updated.

The issue now is that the python system tests are failing for the lilac tests. With a message like this for example...

``` text
Stdout:
ERROR while running:
/glade/work/erik/ctsm_worktrees/external_updates/cime/scripts/create_newcase --output-root /glade/derecho/scratch/erik/tmp/tmp_c8v_9kp/ctsm_build --case /glade/derecho/scratch/erik/tmp/tmp_c8v_9kp/ctsm_build/case --compset I2000Ctsm50NwpSpAsRs --res f10_f10_mg37 --compiler gnu --driver nuopc --project UNSET --run-unsupported --machine ctsm-build --extra-machines-dir /glade/derecho/scratch/erik/tmp/tmp_c8v_9kp/ctsm_build/machine_configuration

Compset longname is 2000_SATM_CLM50%NWP-SP_SICE_SOCN_SROF_SGLC_SWAV
Compset specification file is /glade/work/erik/ctsm_worktrees/external_updates/cime_config/config_compsets.xml
Automatically adding SESP to compset
Compset forcing is 1972-2004
ATM component is Stub atm component
LND component is clm5.0:NWP configuration with satellite phenology:
ICE component is Stub ice component
OCN component is Stub ocn component
ROF component is Stub river component
GLC component is Stub glacier (land ice) component
WAV component is Stub wave component
ESP component is Stub external system processing (ESP) component
Pes     specification file is /glade/work/erik/ctsm_worktrees/external_updates/cime_config/config_pes.xml
Compset specific settings: name is RUN_STARTDATE and value is 2000-01-01
Traceback (most recent call last):
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/scripts/create_newcase", line 18, in <module>
    _main_func()
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/CIME/scripts/create_newcase.py", line 469, in _main_func
    gpu_offload=gpu_offload,
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/CIME/case/case.py", line 2505, in create
    gpu_offload=gpu_offload,
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/CIME/case/case.py", line 1350, in configure
    probed_machine = machobj.probe_machine_name()
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/CIME/XML/machines.py", line 183, in probe_machine_name
    machine = self._probe_machine_name_one_guess(nametomatch)
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/CIME/XML/machines.py", line 213, in _probe_machine_name_one_guess
    return self._probe_machine_name_one_guess_v3(nametomatch)
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/CIME/XML/machines.py", line 306, in _probe_machine_name_one_guess_v3
    return machine
UnboundLocalError: local variable 'machine' referenced before assignment
```

**Review by samsrabin — APPROVED** [review]
I can confirm that `py_env_create` works on both Derecho and Izumi, thanks! I saw the same "local variable 'machine' referenced before assignment" error when I tried the Python system tests on Derecho, but interestingly I didn't see that on Izumi. Maybe a clue?

File: cime_config/testdefs/testlist_clm.xml [hunk 1]
```diff
@@ -1984,10 +1984,8 @@
   <test name="SMS_D_Ld1_Mmpi-serial" grid="f45_f45_mg37" compset="I2000Clm50SpRs" testmods="clm/ptsRLA">
     <machines>
       <machine name="cheyenne" compiler="intel" category="aux_clm"/>
-      <!-- Until Derecho works with intel/Debug use gnu
       <machine name="derecho" compiler="intel" category="aux_clm"/>
       <machine name="derecho" compiler="intel" category="prealpha"/>
-      -->
```
[the remark this answers is not in the document]
Comment (ekluzek) [reply cime_config/testdefs/testlist_clm.xml:1990]: Ahh, good point I can do that. There might only be a few where having both makes sense.

[the remark this answers is not in the document]
Comment (ekluzek) [reply cime_config/testdefs/testlist_clm.xml:1990]: About half of the mpi-serial tests are gnu and half intel. So I alternated between the two. In looking at it I didn't see a reason for any tests to have both intel and gnu.

**Comment by ekluzek** [comment]
Testing is fine. The test

SMS_D_Lm1_Mmpi-serial.CLM_USRDAT.I1PtClm50SpRs.derecho_intel.clm-USUMB_nuopc

is labeled as different answers, but that's just because the baselines don't have all the files. So I consider this fine.

**Comment by ekluzek** [comment]
I thought I had fixed the lilac issue, but in testing for #2416 I see that it's still an issue. So I need to get 

make stest

working again here.

**Comment by ekluzek** [comment]
OK, from talking with @billsacks  it looks like the problem is this one filed in cime.

https://github.com/ESMCI/cime/issues/4588

And @billsacks has a simple fix for that -- that will likely fix the problem coupled with some other simple changes that we talked about.

**Comment by ekluzek** [comment]
This small change to cime solves the issue!

https://github.com/ESMCI/cime/pull/4598

**Comment by ekluzek** [comment]
This also brings up something that we will need to do sometime, but can get away with for now: #2419

**Pull request merged**